src/query/functions/src/scalars/map.rs

```diff
@@ -244,4 +246,50 @@ pub fn register(registry: &mut FunctionRegistry) {
                     .any(|(k, _)| k == key)
             },
         );
+
+    registry.register_3_arg_core::<EmptyMapType, GenericType<0>, GenericType<1>, MapType<GenericType<0>, GenericType<1>>, _, _>(
+        "map_insert",
+        |_, _, _, _| FunctionDomain::Full,
+        |_, key, value, ctx| {
+            let mut b = ArrayType::create_builder(1, ctx.generics);
+            b.put_item((key.into_scalar().unwrap(), value.into_scalar().unwrap()));
+            b.commit_row();
+      
+            return Value::Scalar(MapType::build_scalar(b));
+        },
+    );
+
+    registry.register_passthrough_nullable_3_arg(
+        "map_insert",
+        |_, domian1, _, _| {
+            FunctionDomain::Domain(match domian1 {
+                Some((key_domain, val_domain)) => Some((key_domain.clone(), val_domain.clone())),
```

Comment:
the `key_domain` and `value_domain` need merge with the inserted key and value domain